diff -c -r --new-file v22.2b14/arch.h win/v22.2b14/arch.h
*** v22.2b14/arch.h	2003-12-12 12:08:00.000000000 -0500
--- win/v22.2b14/arch.h	2005-11-03 10:58:20.000000000 -0500
***************
*** 7,12 ****
--- 7,27 ----
  #define ARCH "Sequent DYNIX"
  #endif
  
+ /*
+  * Note - Cygwin runs under windows and automatically
+  * defines a number of windows related things. These
+  * must be undef'd so that the 'normal' windows port code
+  * in MudOS is not used.
+  */
+ #ifdef __CYGWIN__
+ #undef WINNT
+ #undef WIN95
+ #undef WIN98
+ #undef WINSOCK
+ #undef WIN32
+ #define ARCH "Cygwin-32"
+ #endif
+ 
  #ifdef WINNT
  #define ARCH "Microsoft Windows NT"
  #endif
diff -c -r --new-file v22.2b14/build.MudOS win/v22.2b14/build.MudOS
*** v22.2b14/build.MudOS	2003-12-12 12:08:00.000000000 -0500
--- win/v22.2b14/build.MudOS	2005-11-03 11:09:46.000000000 -0500
***************
*** 175,180 ****
--- 175,192 ----
  echo MAKE=$MAKE >Makefile.tmp
  
  #
+ # Determine if running under Cygwin-32 and use a.exe instead
+ # of a.out if so
+ #
+ if test $CYGWIN; then
+   A_OUT=a.exe
+   echo Using a.exe for Cygwin GNU compiler default executable
+ else
+   A_OUT=a.out
+   echo Using standard a.out for compiler default executable
+ fi
+ 
+ #
  # Figure out what to use for CC
  #
  cat >comptest.c <<END
***************
*** 350,356 ****
  }
  END
  $CC $CFLAGS comptest.c
! ARCH=`./a.out`
  
  cat >comptest.c <<END
  int main(int argc, char **argv) { return 0; }
--- 362,368 ----
  }
  END
  $CC $CFLAGS comptest.c
! ARCH=`./$A_OUT`
  
  cat >comptest.c <<END
  int main(int argc, char **argv) { return 0; }
***************
*** 384,408 ****
  #
  # check for some missing functions
  #
! rm a.out
  cat >comptest.c <<END
  int main() {
  strchr(0,0); strrchr(0,0);
  }
  END
  $CC $CFLAGS comptest.c
! if test ! -f a.out; then
      STR="-Dstrchr=index -Dstrrchr=rindex"
  fi
  
! rm a.out
  cat >comptest.c <<END
  int main() {
  memcpy(0,0,0);memset(0,0,0);strtol(0,0,0);strcspn(0,0);
  }
  END
  $CC $CFLAGS comptest.c
! if test ! -f a.out; then
      STRFUNCS=strfuncs.o
  fi
  
--- 396,420 ----
  #
  # check for some missing functions
  #
! rm $A_OUT
  cat >comptest.c <<END
  int main() {
  strchr(0,0); strrchr(0,0);
  }
  END
  $CC $CFLAGS comptest.c
! if test ! -f $A_OUT; then
      STR="-Dstrchr=index -Dstrrchr=rindex"
  fi
  
! rm $A_OUT
  cat >comptest.c <<END
  int main() {
  memcpy(0,0,0);memset(0,0,0);strtol(0,0,0);strcspn(0,0);
  }
  END
  $CC $CFLAGS comptest.c
! if test ! -f $A_OUT; then
      STRFUNCS=strfuncs.o
  fi
  
***************
*** 426,431 ****
--- 438,444 ----
  echo RANLIB=$RANLIB >>Makefile.tmp
  echo A=a >>Makefile.tmp
  echo O=o >>Makefile.tmp
+ echo A_OUT=$A_OUT >>Makefile.tmp
  
  echo "***************** Configuration completed **************"
  echo "Installing MudOS on $ARCH"
diff -c -r --new-file v22.2b14/dead_souls_local_options win/v22.2b14/dead_souls_local_options
*** v22.2b14/dead_souls_local_options	1969-12-31 19:00:00.000000000 -0500
--- win/v22.2b14/dead_souls_local_options	2005-11-03 10:58:24.000000000 -0500
***************
*** 0 ****
--- 1,99 ----
+ #undef SYSMALLOC
+ #define SMALLOC
+ #undef BSDMALLOC
+ #undef WRAPPEDMALLOC
+ #define DEBUGMALLOC
+ #undef SBRK_OK
+ #define DO_MSTATS
+ #define DEBUGMALLOC_EXTENSIONS
+ #define CHECK_MEMORY
+ #undef HAS_STATUS_TYPE
+ #undef COMPRESS_FUNCTION_TABLES
+ #define SANE_EXPLODE_STRING
+ #undef REVERSIBLE_EXPLODE_STRING
+ #undef CAST_CALL_OTHERS
+ #define NONINTERACTIVE_STDERR_WRITE
+ #define NO_LIGHT
+ #undef NO_ADD_ACTION
+ #undef NO_SNOOP
+ #undef NO_ENVIRONMENT
+ #define NO_WIZARDS
+ #undef OLD_TYPE_BEHAVIOR
+ #undef OLD_RANGE_BEHAVIOR
+ #undef OLD_ED
+ #undef SENSIBLE_MODIFIERS
+ #undef CUSTOM_CRYPT
+ #undef COMPAT_32
+ #define STRING_STATS
+ #define ARRAY_STATS
+ #define LOG_CATCHES
+ #undef ARGUMENTS_IN_TRACEBACK
+ #undef LOCALS_IN_TRACEBACK
+ #define MUDLIB_ERROR_HANDLER
+ #define CONFIG_FILE_DIR "./"
+ #define DEFAULT_PRAGMAS PRAGMA_STRICT_TYPES + PRAGMA_WARNINGS + PRAGMA_ERROR_CONTEXT
+ #define SUPPRESS_ARGUMENT_WARNINGS
+ #undef NO_RESETS
+ #define LAZY_RESETS
+ #define SAVE_EXTENSION ".o"
+ #define NO_ANSI
+ #define STRIP_BEFORE_PROCESS_INPUT
+ #define OPCPROF
+ #undef OPCPROF_2D
+ #define TRAP_CRASHES
+ #define THIS_PLAYER_IN_CALL_OUT
+ #define CALLOUT_HANDLES
+ #undef FLUSH_OUTPUT_IMMEDIATELY
+ #define PRIVS
+ #define INTERACTIVE_CATCH_TELL
+ #define RESTRICTED_ED
+ #undef NO_SHADOWS
+ #undef SNOOP_SHADOWED
+ #define RECEIVE_SNOOP
+ #undef PROFILE_FUNCTIONS
+ #undef NO_BUFFER_TYPE
+ #undef BINARIES
+ #define ARRAY_RESERVED_WORD
+ #undef REF_RESERVED_WORD
+ #define PACKAGE_CONTRIB
+ #define PACKAGE_DEVELOP
+ #define PACKAGE_MATH
+ #define PACKAGE_MATRIX
+ #define PACKAGE_MUDLIB_STATS
+ #define PACKAGE_SOCKETS
+ #define PACKAGE_PARSER
+ #define PACKAGE_EXTERNAL
+ #ifdef PACKAGE_EXTERNAL
+ #define NUM_EXTERNAL_CMDS 5
+ #endif
+ #undef PACKAGE_DB
+ #ifdef PACKAGE_DB
+ #undef USE_MSQL 1		/* MiniSQL, it's small; it's free */
+ #define USE_MYSQL 2		/* MySQL, bigger; it's free */
+ #define DEFAULT_DB USE_MYSQL	/* default database */
+ #endif
+ #undef PACKAGE_UIDS
+ #undef AUTO_SETEUID
+ #undef AUTO_TRUST_BACKBONE
+ #undef USE_32BIT_ADDRESSES
+ #define HEARTBEAT_INTERVAL 2000000
+ #define CALLOUT_CYCLE_SIZE 32
+ #define LARGEST_PRINTABLE_STRING 8192
+ #define MESSAGE_BUFFER_SIZE 4096
+ #define APPLY_CACHE_BITS 11
+ #define CACHE_STATS
+ #undef TRACE
+ #undef LPC_TO_C
+ #undef RUNTIME_LOADING
+ #undef TRACE_CODE
+ #define HEART_BEAT_CHUNK      32
+ #undef GET_CHAR_IS_BUFFERED
+ #define SMALL_STRING_SIZE     100
+ #define LARGE_STRING_SIZE     1000
+ #define MAX_SAVE_SVALUE_DEPTH 25
+ #define CFG_MAX_LOCAL_VARIABLES		25
+ #define CFG_EVALUATOR_STACK_SIZE 	1000
+ #define CFG_MAX_CALL_DEPTH		75
+ #define CFG_LIVING_HASH_SIZE		256
+ #define NEXT_MALLOC_DEBUG
+ 
diff -c -r --new-file v22.2b14/efuns_port.c win/v22.2b14/efuns_port.c
*** v22.2b14/efuns_port.c	2003-12-12 12:08:00.000000000 -0500
--- win/v22.2b14/efuns_port.c	2005-11-03 10:58:24.000000000 -0500
***************
*** 115,121 ****
      vec->item[LT_ZONE].subtype = STRING_MALLOC;
      vec->item[LT_ISDST].type = T_NUMBER;
  #if defined(BSD42) || defined(apollo) || defined(_AUX_SOURCE) \
! 	|| defined(OLD_ULTRIX)
      /* 4.2 BSD doesn't seem to provide any way to get these last three values */
      vec->item[LT_GMTOFF].u.number = 0;
      vec->item[LT_ZONE].type = T_NUMBER;
--- 115,121 ----
      vec->item[LT_ZONE].subtype = STRING_MALLOC;
      vec->item[LT_ISDST].type = T_NUMBER;
  #if defined(BSD42) || defined(apollo) || defined(_AUX_SOURCE) \
! 	|| defined(OLD_ULTRIX)|| defined(__CYGWIN__)
      /* 4.2 BSD doesn't seem to provide any way to get these last three values */
      vec->item[LT_GMTOFF].u.number = 0;
      vec->item[LT_ZONE].type = T_NUMBER;
diff -c -r --new-file v22.2b14/GNUmakefile.in win/v22.2b14/GNUmakefile.in
*** v22.2b14/GNUmakefile.in	2003-12-12 12:08:00.000000000 -0500
--- win/v22.2b14/GNUmakefile.in	2005-11-03 10:58:18.000000000 -0500
***************
*** 216,222 ****
  	-rm -f opcodes.h efunctions.h opc.h efun_protos.h
  	-rm -f malloc.c mallocwrapper.c
  	-rm -f func_spec.cpp applies.h applies_table.c files
! 	-rm -f grammar.y comptest* a.out
  	-rm -f packages/Makefile packages/GNUmakefile packages/packages
  
  # remove everything except configuration
--- 216,222 ----
  	-rm -f opcodes.h efunctions.h opc.h efun_protos.h
  	-rm -f malloc.c mallocwrapper.c
  	-rm -f func_spec.cpp applies.h applies_table.c files
! 	-rm -f grammar.y comptest* $(A_OUT)
  	-rm -f packages/Makefile packages/GNUmakefile packages/packages
  
  # remove everything except configuration
diff -c -r --new-file v22.2b14/local_options win/v22.2b14/local_options
*** v22.2b14/local_options	1969-12-31 19:00:00.000000000 -0500
--- win/v22.2b14/local_options	2006-04-05 19:56:06.000000000 -0400
***************
*** 0 ****
--- 1,99 ----
+ #undef SYSMALLOC
+ #define SMALLOC
+ #undef BSDMALLOC
+ #undef WRAPPEDMALLOC
+ #define DEBUGMALLOC
+ #undef SBRK_OK
+ #define DO_MSTATS
+ #define DEBUGMALLOC_EXTENSIONS
+ #define CHECK_MEMORY
+ #undef HAS_STATUS_TYPE
+ #undef COMPRESS_FUNCTION_TABLES
+ #define SANE_EXPLODE_STRING
+ #undef REVERSIBLE_EXPLODE_STRING
+ #undef CAST_CALL_OTHERS
+ #define NONINTERACTIVE_STDERR_WRITE
+ #define NO_LIGHT
+ #undef NO_ADD_ACTION
+ #undef NO_SNOOP
+ #undef NO_ENVIRONMENT
+ #define NO_WIZARDS
+ #undef OLD_TYPE_BEHAVIOR
+ #undef OLD_RANGE_BEHAVIOR
+ #undef OLD_ED
+ #undef SENSIBLE_MODIFIERS
+ #undef CUSTOM_CRYPT
+ #undef COMPAT_32
+ #define STRING_STATS
+ #define ARRAY_STATS
+ #define LOG_CATCHES
+ #undef ARGUMENTS_IN_TRACEBACK
+ #undef LOCALS_IN_TRACEBACK
+ #define MUDLIB_ERROR_HANDLER
+ #define CONFIG_FILE_DIR "./"
+ #define DEFAULT_PRAGMAS PRAGMA_STRICT_TYPES
+ #define SUPPRESS_ARGUMENT_WARNINGS
+ #undef NO_RESETS
+ #define LAZY_RESETS
+ #define SAVE_EXTENSION ".o"
+ #define NO_ANSI
+ #define STRIP_BEFORE_PROCESS_INPUT
+ #define OPCPROF
+ #undef OPCPROF_2D
+ #define TRAP_CRASHES
+ #define THIS_PLAYER_IN_CALL_OUT
+ #define CALLOUT_HANDLES
+ #undef FLUSH_OUTPUT_IMMEDIATELY
+ #define PRIVS
+ #define INTERACTIVE_CATCH_TELL
+ #define RESTRICTED_ED
+ #undef NO_SHADOWS
+ #undef SNOOP_SHADOWED
+ #define RECEIVE_SNOOP
+ #undef PROFILE_FUNCTIONS
+ #undef NO_BUFFER_TYPE
+ #undef BINARIES
+ #define ARRAY_RESERVED_WORD
+ #undef REF_RESERVED_WORD
+ #define PACKAGE_CONTRIB
+ #define PACKAGE_DEVELOP
+ #define PACKAGE_MATH
+ #define PACKAGE_MATRIX
+ #define PACKAGE_MUDLIB_STATS
+ #define PACKAGE_SOCKETS
+ #define PACKAGE_PARSER
+ #undef PACKAGE_EXTERNAL
+ #ifdef PACKAGE_EXTERNAL
+ #define NUM_EXTERNAL_CMDS 5
+ #endif
+ #undef PACKAGE_DB
+ #ifdef PACKAGE_DB
+ #undef USE_MSQL 1		/* MiniSQL, it's small; it's free */
+ #define USE_MYSQL 2		/* MySQL, bigger; it's free */
+ #define DEFAULT_DB USE_MYSQL	/* default database */
+ #endif
+ #undef PACKAGE_UIDS
+ #undef AUTO_SETEUID
+ #undef AUTO_TRUST_BACKBONE
+ #undef USE_32BIT_ADDRESSES
+ #define HEARTBEAT_INTERVAL 1000000
+ #define CALLOUT_CYCLE_SIZE 32
+ #define LARGEST_PRINTABLE_STRING 8192
+ #define MESSAGE_BUFFER_SIZE 4096
+ #define APPLY_CACHE_BITS 11
+ #define CACHE_STATS
+ #undef TRACE
+ #undef LPC_TO_C
+ #undef RUNTIME_LOADING
+ #undef TRACE_CODE
+ #define HEART_BEAT_CHUNK      32
+ #undef GET_CHAR_IS_BUFFERED
+ #define SMALL_STRING_SIZE     100
+ #define LARGE_STRING_SIZE     1000
+ #define MAX_SAVE_SVALUE_DEPTH 25
+ #define CFG_MAX_LOCAL_VARIABLES		25
+ #define CFG_EVALUATOR_STACK_SIZE 	1000
+ #define CFG_MAX_CALL_DEPTH		75
+ #define CFG_LIVING_HASH_SIZE		256
+ #define NEXT_MALLOC_DEBUG
+ 
diff -c -r --new-file v22.2b14/packages/contrib.c win/v22.2b14/packages/contrib.c
*** v22.2b14/packages/contrib.c	2003-12-12 12:07:58.000000000 -0500
--- win/v22.2b14/packages/contrib.c	2005-11-03 11:07:10.000000000 -0500
***************
*** 1050,1058 ****
          break;
      case 'R':
      case 'r':
!         if (!strcasecmp(rel + 1, "oof"))
              found = PLURAL_SUFFIX;
          break;
      case 'S':
      case 's':
          if (!strcasecmp(rel + 1, "niff")) {
--- 1050,1063 ----
          break;
      case 'R':
      case 'r':
!         if (!strcasecmp(rel + 1, "oof")) {
              found = PLURAL_SUFFIX;
          break;
+ 	}
+ 	if (!strcasecmp(rel + 1, "emains")) {
+ 		found = PLURAL_SAME;
+ 		break;
+ 	}
      case 'S':
      case 's':
          if (!strcasecmp(rel + 1, "niff")) {
diff -c -r --new-file v22.2b14/README-CYGWIN win/v22.2b14/README-CYGWIN
*** v22.2b14/README-CYGWIN	1969-12-31 19:00:00.000000000 -0500
--- win/v22.2b14/README-CYGWIN	2005-11-03 11:05:08.000000000 -0500
***************
*** 0 ****
--- 1,8 ----
+ Don't use bash.
+ 
+ In the cmd.exe shell:
+ 
+ 1) cd v22.2b14
+ 2) buildmudos.bat develop
+ 3) make
+ 
diff -c -r --new-file v22.2b14/socket_ctrl.c win/v22.2b14/socket_ctrl.c
*** v22.2b14/socket_ctrl.c	2003-12-12 12:08:00.000000000 -0500
--- win/v22.2b14/socket_ctrl.c	2005-11-03 10:58:24.000000000 -0500
***************
*** 21,27 ****
  #ifdef OLD_ULTRIX
      return fcntl(fd, F_SETOWN, which);
  #else
! #ifdef WINSOCK
      return 1; /* FIXME */
  #else
      return ioctl(fd, SIOCSPGRP, &which);
--- 21,27 ----
  #ifdef OLD_ULTRIX
      return fcntl(fd, F_SETOWN, which);
  #else
! #if defined(WINSOCK) || defined(__CYGWIN__)
      return 1; /* FIXME */
  #else
      return ioctl(fd, SIOCSPGRP, &which);
